src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -288,22 +288,23 @@ async def setup(self, exclude_pool):
                                              {'extra': {'retrieve_children': False}})
         if not dataset or dataset[0]['locked']:
             # Pool is not mounted (e.g. HA node B), temporary set up system dataset on the boot pool
-            self.middleware.logger.debug(
+            self.logger.debug(
                 'Root dataset for pool %r is not available, temporarily setting up system dataset on boot pool',
                 config['pool'],
             )
             self.force_pool = boot_pool
             config = await self.config()
 
-        mounted_pool = None
+        mounted_pool = mounted = None
         for p in psutil.disk_partitions():
             if p.mountpoint == SYSDATASET_PATH:
                 mounted_pool = p.device.split('/')[0]
         if mounted_pool and mounted_pool != config['pool']:
-            self.middleware.logger.debug('Abandoning dataset on %r in favor of %r', mounted_pool, config['pool'])
+            self.logger.debug('Moving system dataset from %r to %r', mounted_pool, config['pool'])
```

Comment:
I believe `abandoning` was a correct word, we are not moving anything, just creating a new dataset on a new pool.

Reply:
We're actually not in a failover scenario. The dataset already exist on the zpool because the other controller migrated it there.

Reply:
Even in a failover scenario what are we moving then?

Reply:
But you're right, `moving` is not the correct word either.